Author: Tollef Fog Heen <tfheen@debian.org>
Description: Snooze a bit to avoid eating too much CPU if you get an SMTP error (Closes: #175173)
Bug-Debian: #175173
Forwarded: https://bugs.launchpad.net/mailman/+bug/966531
Applied-Upstream: 2.1.15


Index: b/Mailman/Queue/OutgoingRunner.py
===================================================================
--- a/Mailman/Queue/OutgoingRunner.py	2010-05-24 17:50:51.000000000 +0200
+++ b/Mailman/Queue/OutgoingRunner.py	2010-05-24 17:50:59.000000000 +0200
@@ -89,6 +89,7 @@
                 syslog('error', 'Cannot connect to SMTP server %s on port %s',
                        mm_cfg.SMTPHOST, port)
                 self.__logged = True
+            self._snooze(0)
             return True
         except Errors.SomeRecipientsFailed, e:
             # Handle local rejects of probe messages differently.
